range.c

```diff
@@ -1268,10 +1286,15 @@ range_minmax(VALUE range)
     if (rb_block_given_p()) {
         return rb_call_super(0, NULL);
     }
-    return rb_assoc_new(
-        rb_funcall(range, id_min, 0),
-        rb_funcall(range, id_max, 0)
-    );
+    VALUE result_min = range_min_internal(0, NULL, range);
+    if (result_min == Qundef) {
+        result_min = rb_funcallv(range, id_min, 0, 0);
+    }
+    VALUE result_max = range_max_internal(0, NULL, range);
+    if (result_max == Qundef) {
+        result_max = rb_funcallv(range, id_max, 0, 0);
+    }
```

Comment:
I think this will run `range_max_internal` twice in the exclusive non-numeric range case.

First, we run it in `minmax`, and it follows the path to return `Qundef`, so we explicitly call `Range#max`. This calls `range_max`, which calls `range_max_internal` again, which returns `Qundef` again, so we call `super` `max`.

I don't know if it is possible, but I think the optimized flow would be to check for `Qundef` and then directly call `super` `max`.

Reply:
Currently, `super` is the only way to directly call a method in middle of the inheritance tree.
Your suggestion is interesting but needs deeper access to the VM internals.
I don't think this is worth doing that much.